docs/guides/integration/coiled.md

```diff
@@ -0,0 +1,155 @@
+- Needing accelerated hardware like GPUs or a big machine with more memory
+- Running the same script with hundreds or thousands of different inputs, in parallel
+
+Coiled makes it simple to run code on cloud hardware.
+
+First, authenticate with Coiled using
+[`coiled login`](https://docs.coiled.io/user_guide/api.html?utm_source=uv-docs#coiled-login) :
+
+```bash
+$ uvx coiled login
+```
+
+You'll be prompted to create a Coiled account if you don't already have one — it's free to start
+using Coiled.
+
+To instruct Coiled to run the script on a virtual machine on AWS, add two comments to the top:
+
+```python title="process.py" hl_lines="1-2"
+# COILED container ghcr.io/astral-sh/uv:debian-slim
+# COILED region us-east-2
+
+# /// script
+# requires-python = ">=3.12"
+# dependencies = [
+#   "pandas",
+#   "pyarrow",
+#   "s3fs",
+# ]
+# ///
+
+import pandas as pd
+
+df = pd.read_parquet(
+    "s3://coiled-data/uber/part.0.parquet",
+    storage_options={"anon": True},
+)
+print(df.head())
+```
+
+!!! tip
+
+    While Coiled supports AWS, GCP, and Azure, this example assumes AWS is being used
+    (see the `region` option below). If you're new to Coiled, you'll automatically have
+    access to a free account running on AWS. If you're not running on AWS, you can either use
+    a valid `region` for your cloud provider or remove the `region` line below.
+
+The comments tell Coiled to use the official [uv Docker image](../integration/docker.md) when
+running the script (ensuring uv is available) and to run in the `us-east-2` region on AWS (where
+this example data file happens to live) to avoid any data egress.
+
+To run the script, use
+[`coiled batch run`](https://docs.coiled.io/user_guide/api.html?utm_source=uv-docs#coiled-batch-run)
+to execute the `uv run` command in the cloud:
+
+```bash hl-lines="1"
+$ uvx coiled batch run \
+    uv run process.py
+```
+
+<!-- TODO
+This command returns immediately, it doesn't wait for the job to finish and it doesn't seem
+like there's a flag for that. It also doesn't happen much faster, because a remote job needs to
+spawn. I also wasn't sure how to get the logs for the job. I eventually found it with
+`uvx coiled batch logs 1067394`. I also tried `uvx coiled batch wait 1067394`, but it didn't have
+an option to show logs and that retrieving the id in the first place was a bit challenging, e.g.,
+`uvx coiled batch status` shows it as a cluster ID at the top but that's not obvious.
+
+I presume some of these problems are because this API is designed around running multiple batch
+jobs, however, if the user experience for running the script locally is that it waits for execution
+and shows the output, then we need to address that difference in user experience here.
+
+It looks like `uvx coiled batch run -- uv run process.py` isn't supported (using the `--` as a
+separator), I wanted to use that for a single-line commmand that still separated the `uv run`
+command.
+-->
```

Comment:
@jrbourbeau Let's talk about this bit and what we can do to clarify this or improve on the experience?